scripts/GetClrDbg.sh

```diff
@@ -76,97 +103,238 @@ generate_nuget_config()
 }
 
-if [ -z "$1" ]; then
-    print_help
-    exit 1
-elif [ "$1" == "-h" ]; then
-    print_help
-    exit 1
-fi
+# Parses and populates the arguments
+parse_and_get_arguments()
+{
+    echo "Parsing Arguments..."
+    while getopts "v:l:srhd" opt; do
+        case $opt in
+            v)
+                __ClrDbgMetaVersion=$OPTARG;
+                ;;
+            l)
+                __InstallLocation=$OPTARG
+                ;;
+            r)
+                __RemoveExisting=true
+                ;;
+            s)
+                __SkipDownloads=true
+                ;;
+            d)
+                __LaunchClrDbg=true
+                ;;
+            h)            
+                print_help
+                exit 1
+                ;;    
+            \?)
+                echo "Error: Invalid Option: -$OPTARG"
+                print_help
+                exit 1;    
+        esac
+    done
+}
 
-# This case statement is done on the lower case version of version_string
-# Add new version constants here
-# 'latest' version may be updated
-# all other version contstants i.e. 'vs2015u2' may not be updated after they are finalized
-version_string="$(echo $1 | awk '{print tolower($0)}')"
-case $version_string in
-    latest)
-        __ClrDbgVersion=14.0.25406-preview-3044032
-        ;;
-    vs2015u2)
-        __ClrDbgVersion=14.0.25406-preview-3044032
-        ;;
-    *)
-        simpleVersionRegex="^[0-9].*"
-        if ! [[ "$1" =~ $simpleVersionRegex ]]; then
-            echo "Error: '$1' does not look like a valid version number."
-            exit 1
-        fi
-        __ClrDbgVersion=$1
-        ;;
-esac
+# Parses and populates the arguments for the legacy commandline.
+get_legacy_arguments()
+{
+    echo "Parsing Legacy Arguments... consider moving to using switches."
+    if [ ! -z "$1" ]; then
+        __ClrDbgMetaVersion=$1
+    fi
+    if [ ! -z "$2" ]; then
+        __InstallLocation=$2
+    fi
+}
 
-echo "Info: Using clrdbg version '$__ClrDbgVersion'"
+# Prints the arguments to stdout for the benefit of the user and does a quick sanity check.
+print_and_verify_arguments()
+{
+    echo "Using arguments"
+    echo "    Version                    : '$__ClrDbgMetaVersion'"
+    echo "    Location                   : '$__InstallLocation'"
+    echo "    SkipDownloads              : '$__SkipDownloads'"
+    echo "    LaunchClrDbgAfter          : '$__LaunchClrDbg'"
+    echo "    RemoveExistingInstallation : '$__RemoveExisting'"
+
+    if [ -z $__ClrDbgMetaVersion ]; then
+        echo "Error: Version is not an optional parameter"
+        exit 1
+    fi
+}
+
+# Converts relative location of the installation directory to absolute location.
+convert_install_path_to_absolute()
+{
+    if [ -z $__InstallLocation ]; then
+        __InstallLocation=$pwd
+    else
+        __InstallLocation=$script_dir/$__InstallLocation
```

Comment:
I think I would suggest first ensuring the directory exists, and then pushd'ing to it. Then you can get the full directory name via pwd.